### PR TITLE
Let random seed setting work on xpu device

### PR DESCRIPTION
**Pull request by retonym**
This Pull Request relates to [Roadmap Issue #1293](https://github.com/pytorch/benchmark/issues/1293) by enhancing our benchmark coverage.

Currently, Torchbench utilizes a custom random seed function that is incompatible with the XPU device backend. 
This incompatibility affects models that include random data augmentation operations, leading to accuracy check failures due to variations in input data across two separate runs.

In this PR, we introduce support for setting a random seed for the XPU backend.

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Review by chuanqi129 — APPROVED** [review]

**Comment by EikanWang** [comment]
@xuzhao9 , I have no privilege to add you to the reviewer list. Could you help review it?

**Comment by xuzhao9** [comment]
@EikanWang Thanks, it looks good to me.

File: torchbenchmark/util/env_check.py [hunk 1]
```diff
@@ -172,9 +172,12 @@ def deterministic_torch_manual_seed(*args, **kwargs):
 
         seed = MAIN_RANDOM_SEED
         import torch.cuda
-
         if not torch.cuda._is_in_bad_fork():
             torch.cuda.manual_seed_all(seed)
+
+        import torch.xpu
```
Comment (xuzhao9) [inline torchbenchmark/util/env_check.py:178]: I have one question, sometimes `torch.xpu` is not available, do we need to wrap it in a try block? cc @retonym

Comment (retonym) [reply torchbenchmark/util/env_check.py:178]: Hi, I tried the code on a non-xpu device, and it works.
The code here is same as the pytorch repo: https://github.com/pytorch/pytorch/blob/main/torch/random.py#L51-L53
